aea/cli/init.py

```diff
@@ -31,15 +31,53 @@
     _update_cli_config,
     pass_ctx,
 )
+from aea.cli.login import do_login
+from aea.cli.register import do_register
+from aea.cli.registry.settings import AUTH_TOKEN_KEY
+from aea.cli.registry.utils import check_is_author_logged_in, is_logged_in
 from aea.configurations.base import PublicId
 
 
+def _registry_init(author):
+    username = author
+
+    if is_logged_in():
+        check_is_author_logged_in(username)
+    else:
+        is_registered = click.confirm("Do you have a Registry account?")
+
+        if is_registered:
+            username = click.prompt("Username", type=str)
+            password = click.prompt("Password", type=str, hide_input=True)
+            do_login(username, password)
+        else:
+            click.echo("Registering a new account...")
+            username = click.prompt("Username", type=str)
+            email = click.prompt("Email", type=str)
+            password = click.prompt("Password", type=str, hide_input=True)
+            password_confirmation = click.prompt(
+                "Confirm password", type=str, hide_input=True
+            )
```

Comment:
It makes sense to do client side validation here of the password confirmation, i.e. while `password_confirmation != password` repeat password_confirmation field?